docs/configuration/configuration.md

```diff
@@ -1878,6 +2046,30 @@ authorization:
   # It is mutually exclusive with `credentials`.
   [ credentials_file: <filename> ]
 
+# Optionally configures OAuth 2.0 authentication using the
+# client credentials grant type. Fetches the token from the
+# specified endpoint using the client access and secret keys.
+# Cannot be used at the same time as basic_auth, authorization, or sigv4.
+oauth2:
+  [ client_id: <string> ]
+  [ client_secret: <secret> ]
+
+  # Read the client secret from a file instead of specifying the value
+  # in `client_secret`.
+  [ client_secret_file: <string> ]
+
+  # Optionally specify scopes for the token.
+  scopes:
+    [ - <string> ... ]
+
+  # The URL to fetch the token from.
+  [ token_url: <string> ]
+
+  # Optional parameters to append to the token URL.
+  # Takes the form of `- name: value`.
```

Comment:
You already specify the form, it is not needed to explicitly write it.

also, this is not - name: value, it should be {name: value}